src/core/gridstate_mod.F90

```diff
@@ -0,0 +1,145 @@
+!!! coments for GridState_Mod.F90
+
+
+module GridState_Mod
+
+   USE Error_Mod
+   USE precision_mod
+
+   IMPLICIT NONE
+   PRIVATE
+
+   PUBLIC :: Grid_Init_State
+   PUBLIC :: Grid_State_Allocate
+   PUBLIC :: Grid_Cleanup
+
+   type, public :: GrdState
```

Comment:
Why not "GridState"?

Reply:
And really "GridStateType" would be better, so that you can call the instance or dummy parameter "GridState"...